src/cyclotom.c

```diff
@@ -143,7 +143,7 @@ static inline void SET_NOF_CYC(Obj cyc, Obj val)
 #define XXX_CYC(cyc,len)        (EXPOS_CYC(cyc,len)[0])
 
 
-static ModuleStateOffset CycStateOffset = -1;
+static UInt CycStateOffset;
```

Comment:
I should point out that changing these offsets to UInt looses the extra sanity check of initializing them to -1, which we used to detect modules that forgot to init the offset. But:
1. we could just get that back by setting to `~(UInt)0` or just any value larger than the maximal allowed offset (using some `MAX_UINT_FOOBAR` value from the `limits.h` or so);
2. we could add 1 to all offsets making 0 invalid (and subtracting 1 again when using the offset);
3. we could also just not care (as this PR does), thinking that now that we register the module state and initialize the offsets in a uniform way, it's far less likely that this gets "forgotten", and anyway, we don't worry about this for countless other things in the kernel...

Reply:
Stupid question, why not make it an Int, then you can keep -1? These offsets are never going to be need an unsigned type.

This might be a place where we disagree, I always want to make things an Int unless they have to be unsigned, whereas I know some people like UInt for things which should be unsigned.